src/backend/joanie/core/api/client/__init__.py

```diff
@@ -541,6 +542,59 @@ def withdraw(self, request, pk=None):  # pylint: disable=no-self-use, invalid-na
 
         return Response(status=HTTPStatus.NO_CONTENT)
 
+    @extend_schema(
+        request=None,
+        responses={
+            (200, "application/json"): OpenApiTypes.OBJECT,
+            404: serializers.ErrorResponseSerializer,
+            422: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(detail=True, methods=["POST"])
+    def submit_installment_payment(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Pay a failed installment that was scheduled for a given order.
+        """
+        order = self.get_object()
+        if order.state != enums.ORDER_STATE_FAILED_PAYMENT:
+            return Response(
+                {
+                    "detail": "The order is not in failed payment state for last payment."
+                },
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        payment_backend = get_payment_backend()
+        installment = order.get_first_installment_refused()
```

Comment:
If no installment found you should abort the request by returning a 400 saying that there is nothing to pay.

Reply:
Okay, but that would mean there is an issue with the order's state, and the installment state.

As far as I've understood the new workflow, the order's state become 'ORDER_STATE_FAILED_PAYMENT' when there is 1 installment that has the state 'PAYMENT_STATE_REFUSED'. 
I've put already a safety net in the condition above to only treat orders in the state of 'ORDER_STATE_FAILED_PAYMENT'.

So either I : 
- take away the first condition to allow only treatment on order.state == "ORDER_STATE_FAILED_PAYMENT",
- take away the second condition where "installment" returns None, meaning no installment found.

Which one should i keep ?

Reply:
Keep both. Shit may happen and there is no strong constraint between bot state.

Reply:
Good, thanks let's do that